### PR TITLE
Adds loading indicator for current location API call

### PR DESCRIPTION
**Pull request by isaacchansky**
Extends the findPollingLocationFor function w/ optional success and
failure callbacks.
Utilizing these callbacks to modify the button text & state within the
click handler to indicate that the app is doing stuff.

Note: I'm not currently using the error callback for anything, but this seems like a nice, familiar pattern we could follow (passing callback parameters).

Could also use promises just as easily, but I figured this pattern is more in style with the current codebase.

### CODE REVIEW

**Comment by isaacchansky** [comment]
This is a quick, simple attempt to remedy #33 

I'm not sure how people generally feel about the "turn a button into the loader" pattern, and if we should have some motion involved, but this is just a first pass. 


See screenshot below for what it looks like after you click "Use current location":

screen shot 2016-09-13 at 10 49 21 pm

**Comment by carpeliam** [comment]
I think callbacks are the right way to go at this point. As much as I'd prefer promises, I think the ratio of required work to payoff just isn't worth it, considering there are so many other architectural changes that'd be more beneficial. Perhaps the callbacks should be passed an arg? It's somewhat common to let the successCallback know what the results were, and to let the errorCallback know what the error was... but we'd probably end up dropping that on the floor anyway, so I don't really care so much there.

I'll let @thecristen (and perhaps @WheresHJ?) weigh in on the design aspects first..

File: app/scripts/polling_location_finder.js [hunk 1]
```diff
@@ -78,7 +78,9 @@ define(['jquery', 'geojson', 'json!vendor/ELECTIONS_WardsPrecincts.geojson', 'js
         return encodeURI(url + destination);
     }
 
-    return function(latLng) {
+    // successCallback & errorCallback are optional parameters,
+    // which will be executed on success or failure of the google maps direction service, respectively
```
Comment (carpeliam) [inline app/scripts/polling_location_finder.js]: I think you can do without the comments- I think "callback" in the variable name is self-documenting, and the `if` statements make it clear that they're optional (and this is a common enough pattern).

**Comment by thecristen** [comment]
I happen to quite like the button-as-loader pattern! So yay. I totally would love to see motion added to this, for example with a loader from http://tobiasahlin.com/spinkit/ - I like the second loading icon on this page (and it's CSS only!)

**Comment by isaacchansky** [comment]
Updated!
vote-camb-loader

**Pull request merged**